### PR TITLE
Add a filter to the cms type specials so it can be filtered on the sp…

### PR DESCRIPTION
**Pull request by royderks**
…ecial type

### CODE REVIEW

File: src/es/cms/utils.ts [hunk 1]
```diff
@@ -178,4 +178,27 @@ function getDateFilter(types: Array<string> | null, filters: Array<FilterInput>)
...
 }
 
-export { getSearchQuery, getThemeFilter, getDateFilter }
+// Constructs the ES query to filter on subtype
+function getSubTypeFilter(filters: Array<FilterInput>) {
+  const subTypeFilter = filters.find((filter) => filter.type === FILTERS['SUBTYPE'].type)
+
+  if (subTypeFilter && subTypeFilter.values) {
+    return subTypeFilter.values
+      .map((value: any) => {
```
Comment (jonkoops) [inline src/es/cms/utils.ts:187]: Looking at the implementation `value` is expected to be of type `string`.

Comment (jonkoops) [inline src/es/cms/utils.ts:187]: I think we could use `flatMap` here instead of `map` and then `flat`.

File: src/graphql/search/cms/index.ts [hunk 2]
```diff
@@ -26,21 +29,30 @@ const cmsSearch = (type: string) => async (
     types: [type],
     filters: filterInput,
     sort,
+    subType,
   })
 
-  const cmsFilters: any = await loaders.cms.load(
+  const formattedResults = getFormattedResults(results)
+
+  const cmsThemeFilters: any = await loaders.cms.load(
```
Comment (jonkoops) [inline src/graphql/search/cms/index.ts:37]: Can we use another type other than `any` here? Perhaps like `subTypeFilters` below here.

File: src/graphql/search/cms/normalize.ts [hunk 3]
```diff
@@ -142,6 +151,21 @@ function getDateFilterOptions(): Array<FilterOptions> {
...
+  filters: Array<ESFilters>,
+  subTypeCount?: Array<SubTypeFilterCount>,
+): Array<FilterOptions> {
+  return filters.map(({ key }) => {
+    const { count } = subTypeCount?.find((count) => count.key === key) || {}
+
+    return {
+      id: `${FILTERS['SUBTYPE'].type}:${key}`,
+      label: getCapitalizedString(key),
+      count,
```
Comment (jonkoops) [inline src/graphql/search/cms/normalize.ts:164]: Count is allowed to be undefined?

File: src/es/cms/utils.ts (hunk 1 above)
Comment (royderks) [reply src/es/cms/utils.ts]: This typing was not even needed, as it could be inherit from above

**Pull request merged**